grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1757,11 +1757,13 @@ class JsonController {
                     """
                         SELECT ps.id, ps.code, ps.supplier_code, ps.name, ps.manufacturer_code, ps.manufacturer_id from product_supplier ps
                         LEFT OUTER JOIN product_supplier_preference default_preference ON default_preference.product_supplier_id = ps.id AND default_preference.destination_party_id IS NULL
+                        LEFT OUTER JOIN preference_type default_preference_type ON default_preference.preference_type_id = default_preference_type.id
                         LEFT OUTER JOIN  product_supplier_preference preference ON preference.product_supplier_id = ps.id AND preference.destination_party_id = :destinationPartyId
+                        LEFT OUTER JOIN preference_type party_preference_type ON preference.preference_type_id = party_preference_type.id
                         WHERE product_id = :productId
                         AND supplier_id = :supplierId
-                        AND CASE WHEN preference.id IS NOT NULL THEN preference.preference_type_id != 4 ELSE
-                        (CASE WHEN default_preference.id IS NOT NULL THEN default_preference.preference_type_id != 4 ELSE true END) END
+                        AND CASE WHEN preference.id IS NOT NULL THEN party_preference_type.validation_code != "HIDE" ELSE
```

Comment:
Pass it in params as `:validationCode` and `validationCode: ValidationCode.HIDE`